test/extended/networking/commatrix.go

```diff
@@ -0,0 +1,215 @@
+package networking
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift-kni/commatrix/pkg/client"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/types"
+	"github.com/openshift-kni/commatrix/pkg/utils"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/sirupsen/logrus"
+
+	commatrixcreator "github.com/openshift-kni/commatrix/pkg/commatrix-creator"
+	"github.com/openshift-kni/commatrix/pkg/endpointslices"
+
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var _ = g.Describe("[sig-network][Feature:commatrix][Serial]", func() {
+	g.It("should apply firewall by blocking all ports except the ones OCP is listening on", func() {
+		g.By("generating the commatrix")
+
+		cs, err := client.New()
+		if err != nil {
+			panic(fmt.Errorf("failed creating the k8s client: %w", err))
+		}
+
+		deploymentstr := "deployment"
+		isSNO, err := isSNOCluster()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isSNO {
+			deploymentstr = "sno"
+		}
+		deployment, _ := types.GetDeployment(deploymentstr)
+
+		epExporter, err := endpointslices.New(cs)
+		if err != nil {
+			panic(fmt.Errorf("failed creating the endpointslices exporter: %w", err))
+		}
+
+		commMatrix, err := commatrixcreator.New(epExporter, "", "", types.Baremetal, deployment)
+		if err != nil {
+			panic(fmt.Errorf("failed creating comm matrix creator: %w", err))
+		}
+
+		matrix, err := commMatrix.CreateEndpointMatrix()
+		if err != nil {
+			panic(fmt.Errorf("failed creating endpoint matrix: %w", err))
+		}
+
+		masterMat, workerMat := separateMatrixByRole(*matrix)
+
+		masterNFT, err := masterMat.ToNFTables()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		workerNFT, err := workerMat.ToNFTables()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("apply firewall on master nodes")
+		err = applyRulesToNode(cs, "master", masterNFT)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		if !isSNO {
+			g.By("apply firewall on worker nodes")
+			err = applyRulesToNode(cs, "worker", workerNFT)
+			o.Expect(err).ToNot(o.HaveOccurred())
+		}
+	})
+})
+
+func applyRulesToNode(cs *client.ClientSet, role string, NFTtable []byte) error {
+	utilsHelpers := utils.New(cs)
+	ns := consts.DefaultDebugNamespace + "firewall"
```

Comment:
Why do you use  consts.DefaultDebugNamespace + "firewall", can you use the same ns? or just set here the ns you want :)

Reply:
hh yah i wanted to use with firewall name so i changed now